pkg/search/postgres/common.go

```diff
@@ -422,18 +535,72 @@ func standardizeQueryAndPopulatePath(q *v1.Query, schema *walker.Schema, queryTy
 			parsedQuery.Data = append(parsedQuery.Data, queryEntry.Having.Values...)
 		}
 	}
-	populateGroupBy(parsedQuery, schema, dbFields)
+
+	if err := populateGroupBy(parsedQuery, q.GetGroupBy(), schema, dbFields); err != nil {
+		return nil, err
+	}
 	if err := populatePagination(parsedQuery, q.GetPagination(), schema, dbFields); err != nil {
 		return nil, err
 	}
 	if err := applyPaginationForSearchAfter(parsedQuery); err != nil {
 		return nil, err
 	}
 
-	// Populate primary key select fields once so that we do not have to evaluate.
+	// Populate primary key select fields once so that we do not have to evaluate multiple times.
 	parsedQuery.populatePrimaryKeySelectFields()
 
-	// Nothing more to do. Currently, GROUP BY clause is only supported on primary keys.
+	return parsedQuery, nil
+}
+
+func standardizeSelectQueryAndPopulatePath(q *v1.Query, schema *walker.Schema, queryType QueryType) (*query, error) {
+	nowForQuery := time.Now()
+	standardizeFieldNamesInQuery(q)
+	innerJoins, dbFields := getJoinsAndFields(schema, q)
+
+	if q.GetSelect() == nil && q.GetQuery() == nil {
+		return nil, nil
+	}
+	if q.GetSelect() == nil {
+		return nil, errors.New("select portion of the query cannot be empty")
+	}
+
+	parsedQuery := &query{
+		Schema:     schema,
+		QueryType:  queryType,
+		From:       schema.Table,
+		InnerJoins: innerJoins,
+	}
+
+	err := populateSelect(parsedQuery, schema, q.GetSelect(), dbFields)
+	if err != nil {
+		return nil, err
+	}
+
+	queryEntry, err := compileQueryToPostgres(schema, q, dbFields, nowForQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	if queryEntry != nil {
+		parsedQuery.Where = queryEntry.Where.Query
+		parsedQuery.Data = queryEntry.Where.Values
+		// We won't need this once highlights is removed and fields can only be selected when explicitly specified in the query.
```

Comment:
Might be good to add a TODO here or something else easily identifiable so we know to come back and clean this up